modules/siddhi-core/src/main/java/io/siddhi/core/table/holder/ListEventHolder.java

```diff
@@ -44,13 +46,14 @@ public ListEventHolder(StreamEventFactory tableStreamEventFactory, StreamEventCo
     }
 
     @Override
-    public void add(ComplexEventChunk<StreamEvent> addingEventChunk) {
+    public int add(ComplexEventChunk<StreamEvent> addingEventChunk) {
         addingEventChunk.reset();
         while (addingEventChunk.hasNext()) {
             ComplexEvent complexEvent = addingEventChunk.next();
             StreamEvent streamEvent = tableStreamEventFactory.newInstance();
             eventConverter.convertComplexEvent(complexEvent, streamEvent);
             this.add(streamEvent);
         }
+        return findEventChunkSize(addingEventChunk.getFirst());
```

Comment:
This is a very costly operation finding the event size every time, can't we do any other optimizations